main.go

```diff
@@ -0,0 +1,295 @@
+package main
+
+import (
+	"bytes"
+	"errors"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"crypto/ecdsa"
+	"github.com/dgrijalva/jwt-go"
+
+	"github.com/codahale/blake2"
+	"github.com/gomodule/redigo/redis"
+	"github.com/tidwall/resp"
+)
+
+var jwtPubKey *ecdsa.PublicKey
+
+type ZdbBackend struct {
+	Server redis.Conn
+}
+
+type Session struct {
+	Authenticated bool
+	Username      string
+	Backend       *ZdbBackend
+}
+
+type Sessions struct {
+	List    map[*resp.Conn]*Session
+	Backend *ZdbBackend
+}
+
+//
+// its you online
+//
+const (
+	iyoPubKey = `-----BEGIN PUBLIC KEY-----
+MHYwEAYHKoZIzj0CAQYFK4EEACIDYgAES5X8XrfKdx9gYayFITc89wad4usrk0n2
+7MjiGYvqalizeSWTHEpnd7oea9IQ8T5oJjMVH5cc0H5tFSKilFFeh//wngxIyny6
+6+Vq5t5B0V0Ehy01+2ceEon2Y0XDkIKv
+-----END PUBLIC KEY-----`
+)
+
+func cryptoInit() {
+	var err error
+
+	jwtPubKey, err = jwt.ParseECPublicKeyFromPEM([]byte(iyoPubKey))
+	if err != nil {
+		log.Panicf("failed to parse pub key:%v\n", err)
+	}
+}
+
+//
+// list of sessions
+//
+func NewSessions(backend *ZdbBackend) *Sessions {
+	return &Sessions{
+		List:    make(map[*resp.Conn]*Session),
+		Backend: backend,
+	}
+}
+
+// single session
+func (ss *Sessions) NewSession() *Session {
+	return &Session{
+		Authenticated: false,
+		Username:      "",
+		Backend:       ss.Backend,
+	}
+}
+
+// create a new Session based on a new connection
+func (ss *Sessions) SessionAccept(conn *resp.Conn) bool {
+	log.Printf("Creating a new session for [%v]\n", conn.RemoteAddr)
+	ss.List[conn] = ss.NewSession()
+	return true
+}
+
+// verify a jwt token is valid
+func verifyToken(tokenStr string) (bool, string, error) {
+	token, err := jwt.Parse(tokenStr, func(token *jwt.Token) (interface{}, error) {
+		if token.Method != jwt.SigningMethodES384 {
+			return nil, fmt.Errorf("Unexpected signing method: %v", token.Header["alg"])
+		}
+		return jwtPubKey, nil
+	})
+
+	if err != nil {
+		return false, "", err
+	}
+
+	if claims, ok := token.Claims.(jwt.MapClaims); ok && token.Valid {
```

Comment:
How exactly are you verifying anything here ?
I don't see you comparing value in the claims to anything.

Reply:
According to the documentation, this is how to validate a jwt.